### PR TITLE
Replace template components with spin-off

### LINKED ISSUE #11105: Spin off template components into their own package
Wagtail's [template components](https://docs.wagtail.org/en/stable/extending/template_components.html) module has been identified as a good candidate for spinning off into a standalone package - it's proved broadly useful across Wagtail and would be of use to the wider Django ecosystem, and the API has remained mostly stable over the last few releases. Spinning it out into its own package would encourage better separation of concerns, allow it to be better tested and documented, and would generally reduce the "walled garden" effect of Wagtail having its own internal conventions not seen elsewhere.

Further down the line, this could be followed up by releasing the table UI component as another standalone package.

### PR DESCRIPTION
**Pull request by tbrlpld**
Fixes #11105 

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

This PR replaces the "template components" previously found in `wagtail.admin.ui.components` and the related template tags in `wagtail.admin.templatetags.wagtailadmin_tags` with imports from the [Laces](https://github.com/tbrlpld/laces) package. Laces was created to make the "template components" idea available to the wider Django eco-system. 

To avoid the need to update all usage locations in Wagtail and save people from having to add Laces to their installed apps, the previous implementations are replaced with imports from Laces.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by lb-** [comment]
Thanks @tbrlpld - I have flagged for review. Looks pretty straight forward, just wondering if it's worth us keeping a simple set of `TestComponentTag` tests. While we can trust this package we do want to also ensure that the basics keep working with our wrapper.

It might be best for @gasman  to take a look at this one, I have requested a review.

**Comment by tbrlpld** [comment]
> Thanks @tbrlpld - I have flagged for review. Looks pretty straight forward, just wondering if it's worth us keeping a simple set of `TestComponentTag` tests. While we can trust this package we do want to also ensure that the basics keep working with our wrapper.
> 
> It might be best for @gasman to take a look at this one, I have requested a review.

Happy to bring the template tag tests back if that is desired.

**Comment by tbrlpld** [comment]
> As part of the adoption of this, I know there will be a mention in the release notes, but will there also be a deprecation timeline to drop the shadowed names and require implementers to switch to the new paths?

I am not aware of any deprecation plans for the internal names, but maybe someone else can say more here.

**Comment by lb-** [comment]
I think for now we would keep the internal Wagtail names as part of this PR, if we do consider a deprecation process we would probably start that closer to a 7.0 release. As with any rename, there would be ample time across multiple releases to change any code that uses this.

For this specific case, it may make sense to always keep the template name redirects for convenience.

**Comment by gasman** [comment]
If we were to switch to using the laces tag library directly rather than 'proxying' it through wagtailadmin_tags, then we'd need everyone to add `laces` to INSTALLED_APPS - I don't think that's desirable for something that's only used internally within Wagtail.

**Comment by gasman** [comment]
Merged in 10c1e12285929624aab20207f8afb609b850a886

File: setup.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "openpyxl>=3.0.10,<4.0",
     "anyascii>=0.1.5",
     "telepath>=0.1.1,<1",
+    "laces>=0.1,<0.2",
```
[the remark this answers is not in the document]
Comment (tbrlpld) [reply setup.py:39]: Yea, that is actually a good call to open dependencies up more. Happy to submit that PR once there is a new Laces version :)

**Pull request closed without merging**